lib/stdlib/test/property_test/binary_prop.erl

```diff
@@ -0,0 +1,18 @@
+-module(binary_prop).
+
+-export([prop_hex_encode_2/0]).
+
+-compile([export_all, nowarn_export_all]).
+
+-include_lib("common_test/include/ct_property_test.hrl").
+
+prop_hex_encode_2() ->
+    ?FORALL(Data, data(),
+            begin
+                UpperHex = binary:encode_hex(Data, uppercase),
+                LowerHex = binary:encode_hex(Data, lowercase),
+                binary:decode_hex(LowerHex) =:= Data andalso binary:decode_hex(UpperHex) =:= Data
```

Comment:
This is nice as far as it goes, and you should keep it, but this only tests that what was hex-encoded can be hex-decoded.  In other words, it is not tested if `encode_hex` produces the correct output, only that `decode_hex` can decode whatever `encode_hex` produced back into the initial binary.

A way you could test correct output (without repeating the actual implementation) could be with a helper function like this:
```erlang
check_hex_encoded(<<I1:4, I2:4, Ins/binary>>, <<U1:8, U2:8, UCs/binary>>, <<L1:8, L2:8, LCs/binary>>) ->
    check_hex_chars_match(I1, U1, L1) andalso
    check_hex_chars_match(I2, U2, L2) andalso
    check_hex_encoded(Ins, UCs, LCs);
check_hex_encoded(<<>>, <<>>, <<>>) ->
    true;
check_hex_encoded(_, _, _) ->
    false.

check_hex_chars_match(X, U, L) when X < 10 ->
    (U =:= $0 + X) andalso (L =:= $0 + X);
check_hex_chars_match(X, U, L) ->
    (U =:= $A + X -10) andalso (L =:= $a + X -10).
```
... and call it from the property like `check_hex_encoded(Data, UpperHex, LowerHex)`.
*(Note that I just typed this off the top of my head, didn't test it, so there may be bugs and/or typos :sweat_smile:)*

I admit this won't be hideously fast (for which reason I would just use the normally-growing `binary()` generator instead of resizing it), but with tests it is more important to be thorough than to be fast :wink:

Reply:
Done